File: src/test/java/com/github/rssanders3/spark/java/MainTest.java
package com.github.rssanders3.spark.java;

import junit.framework.TestCase;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.SQLContext;

/**
 * Created by robertsanders on 11/9/16.
 */
public class MainTest extends TestCase {

    private String master = "local[2]";
    private String appName = this.getClass().getSimpleName();

    private SparkConf conf = new SparkConf().setMaster(master).setAppName(appName);
    private JavaSparkContext sc;
    private SQLContext sqlContext;

    @Override
    protected void setUp() throws Exception {
        super.setUp();
        sc = new JavaSparkContext(conf);
        sqlContext = new SQLContext(sc);
    }

    @Override
    protected void tearDown() throws Exception {
        if (sc != null) {
            sc.stop();
            sc = null;
        }
        super.tearDown();
    }

    public void test() {
        //Main.{your_function}
        //assertTrue("Message if assert fails", true);
        //assertEquals("Message if assert fails", true, true);
    }

}

File: src/test/python/MainTest.py
__author__ = 'robertsanders'

from pyspark import SparkConf, SparkContext
from pyspark.sql import SQLContext
import unittest


class MainTest(unittest.TestCase):

    master = "local[2]"

    conf = SparkConf().setAppName(__name__).setMaster(master)
    sc = None
    sqlContext = None

    @staticmethod
    def setUp(**kwargs):
        sc = SparkContext(conf)
        sqlContext = SQLContext(sc)

    @staticmethod
    def tearDown(**kwargs):
        if sc != None:
            sc.stop()
            sc = None

    def test_contexts_not_none(self):
        self.assertNotEquals(None, sc)
        self.assertNotEquals(None, sqlContext)

if __name__ == '__main__':
    unittest.main()